src/device/scrollCursorIntoView.ts

```diff
@@ -57,14 +60,15 @@ const scrollCursorIntoView = () => {
   // otherwise Safari scrolls to the top after MultiGesture
   // See: touchmove in MultiGesture.tsx
   if (window.scrollY === 0 && isTouch && isSafari()) {
+    preventScrollDetection()
     window.scrollBy(0, 1)
   }
 
-  // Wait for the next render as this function is called immediately after an action is dispatched.
-  // An animation frame should be enough time.
-  requestAnimationFrame(() => {
+  // Wait for 100ms as this function is called immediately after an action is dispatched.
+  // An animation frame is not enough time if multiple levels of thoughts are being pulled.
+  setTimeout(() => {
     scrollIntoViewIfNeeded(document.querySelector('.editing'))
-  })
+  }, 100)
```

Comment:
Can we add the 100ms only when needed? 100ms is going to cause a noticeable delay in a lot of other cases.